src/pds/ingress/pds-nucleus-product-completion-checker.py

```diff
@@ -0,0 +1,321 @@
+"""
+=========================================
+pds-nucleus-product-completion-checker.py
+=========================================
+
+Lambda function to check if the staging S3 bucket has received a complete product 
+with all required file. This lambda function is triggered by S3 bucket events
+when a new file is copied to the staging S3 bucket.
+
+"""
+
+import json
+import urllib.parse
+import shutil
+import boto3
+import os
+from xml.dom import minidom
+from boto3.dynamodb.conditions import Key, Attr
+
+print('Loading pds-nucleus-product-completion-checker function')
+
+s3 = boto3.client('s3')
+dynamodb = boto3.resource('dynamodb')
+
+# Main lambda handler
+def lambda_handler(event, context):
+
+    # Get the object from the event and show its content type
+    bucket = event['Records'][0]['s3']['bucket']['name']
+    key = urllib.parse.unquote_plus(event['Records'][0]['s3']['object']['key'], encoding='utf-8')
+    s3_url_of_data_file = "s3://" + bucket + "/" + key
+    s3_url_of_product_label = "s3://" + bucket + "/" + key
+    efs_mount_path = os.environ.get('EFS_MOUNT_PATH')
+    print("Lambda Request ID:", context.aws_request_id)
+
+    try:
+
+        # Product label received
+        if s3_url_of_data_file.lower().endswith(".xml"):
+            
+            missing_files = handle_missing_files_for_product_label(s3_url_of_product_label, bucket, key)
+
+            copy_file_to_efs(bucket, key, efs_mount_path)
+
+            # If there are any missing files for the given product label
+            if (len(missing_files) > 0):
+                save_incomplete_product(s3_url_of_product_label, missing_files)
+            else:
+                product_completed(s3_url_of_product_label)
+                
+        # Data file received
+        else:
+            
+            if not s3_url_of_data_file.lower().endswith("/"): # Not a directory
+                copy_file_to_efs(bucket, key, efs_mount_path)
+                save_received_file(s3_url_of_data_file)
+                update_existing_incomplete_products(s3_url_of_data_file)
+                delete_expected_file(s3_url_of_data_file)
+            
+        return 'Copied'
+    except Exception as e:
+        print(e)
+        # print('Error getting object {} from bucket {}. Make sure they exist and your bucket is in the same region as this function.'.format(key, bucket))
```

Comment:
I really don't like to see commented-out code without an explanatory comment saying why it was commented out. I would prefer to add the explanation—or just delete the line outright.

Reply:
Removed comment